tests/unit/test_define_xml_reader.py

```diff
@@ -290,3 +290,12 @@ def test_extract_variable_metadata_not_found():
         reader = DefineXMLReaderFactory.from_file_contents(contents)
         with pytest.raises(DomainNotFoundInDefineXMLError):
             reader.extract_variables_metadata(domain_name="not found domain")
+
+
+def test_get_domain_key_sequence():
+    with open(test_define_file_path, "rb") as file:
+        contents: bytes = file.read()
+        reader = DefineXMLReaderFactory.from_file_contents(contents)
+        ec_key_sequence: dict = reader.get_domain_key_sequence(domain_name="EC")
+        print(ec_key_sequence)
+        assert ec_key_sequence == ["STUDYID", "USUBJID", "ECTRT", "ECSTDTC", "ECREASOC"]
```

Comment:
Nice, it looks like this is doing EC how I would expect. I think you should also add a test for a SUPP dataset as well.
For example, the SUPPEC keys in this define.xml should be:
STUDYID, RDOMAIN, USUBJID, IDVAR, IDVARVAL, QNAM

Reply:
Good call, this does need some updating to work with supp datasets